lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/resourcemanager/ResourceEvent.java

```diff
@@ -58,4 +60,9 @@
    * @return Runtime name of the resource
    */
   String getRuntimeName();
+
+  /**
+   * @return node label expression of the resource
+   */
+  Optional<Set<String>> getNodeLabelExpressions();
```

Comment:
YARN returns `null` if there is no label on a node. For consistency, We'd better to use `Optional` rather than empty `Set`.